src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -928,13 +964,17 @@ private void EnterParameter(ParameterSymbol parameter, TypeSymbolWithAnnotations
             {
                 if (EmptyStructTypeCache.IsTrackableStructType(parameterType.TypeSymbol))
                 {
-                    InheritNullableStateOfTrackableStruct(parameterType.TypeSymbol, slot, valueSlot: -1, isByRefTarget: parameter.RefKind != RefKind.None, slotWatermark: GetSlotWatermark());
+                    InheritNullableStateOfTrackableStruct(
+                        parameterType.TypeSymbol,
+                        slot,
+                        valueSlot: -1,
+                        isDefaultValue: parameter.ExplicitDefaultConstantValue?.IsNull == true,
```

Comment:
>parameter.ExplicitDefaultConstantValue?.IsNull == true, [](start = 40, length = 55)

Does that work for `void M(T t = default)`?

Reply:
We're only interested in cases where `parameter.Type` is a trackable `struct`.

---
In reply to: [246931559](https://github.com/dotnet/roslyn/pull/32304#discussion_r246931559) [](ancestors = 246931559)